File: AIModelsEvaluation/GPT-OSS/frontend/src/domain/Order.ts
import { OrderItem } from "./OrderItem";
import { Customer } from "./Customer";

export enum OrderStatus {
  NEW = "NEW",
  ACCEPTED = "ACCEPTED",
  SHIPPED = "SHIPPED",
  CLOSED = "CLOSED",
}

export interface Order {
  id: string;
  customer: Customer;
  items: OrderItem[];
  totalAmount: number;
  status: OrderStatus;
  createdAt: string;
  updatedAt: string;
  invoiceId?: string;
}

File: method_pipeline/generated_test/GPT-OSS/chatdev-030726-QA2/sdk_chatdev_20260703002320_20260703002320/code_workspace/app/schemas/order.py
"""
Pydantic schemas for Order and OrderLineItem.
"""

from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime
from enum import Enum

class OrderStatus(str, Enum):
    PLACED = "placed"
    ACCEPTED = "accepted"
    REJECTED = "rejected"
    INVOICED = "invoiced"
    PAID = "paid"
    SHIPPED = "shipped"
    CLOSED = "closed"

class OrderLineItemBase(BaseModel):
    product_id: int
    quantity: int = 1
    unit_price: float

class OrderLineItemCreate(OrderLineItemBase):
    pass

class OrderLineItemOut(OrderLineItemBase):
    id: int
    class Config:
        orm_mode = True

class OrderBase(BaseModel):
    customer_id: int
    status: OrderStatus = OrderStatus.PLACED
    total_amount: float = 0.0

class OrderCreate(OrderBase):
    line_items: List[OrderLineItemCreate]

class OrderUpdate(BaseModel):
    status: Optional[OrderStatus] = None
    total_amount: Optional[float] = None

class OrderOut(OrderBase):
    id: int
    created_at: datetime
    updated_at: datetime
    line_items: List[OrderLineItemOut] = []
    class Config:
        orm_mode = True
